examples/stake-tokens/components/types.tsx

```diff
@@ -0,0 +1,171 @@
+import { MouseEventHandler, ReactNode, RefObject } from 'react';
+import { IconType } from 'react-icons';
+
+export interface ChooseChainInfo {
+  chainName: string;
+  chainRoute?: string;
+  label: string;
+  value: string;
+  icon?: string;
+  disabled?: boolean;
+}
+
+export enum WalletStatus {
+  NotInit = 'NotInit',
+  Loading = 'Loading',
+  Loaded = 'Loaded',
+  NotExist = 'NotExist',
+  Rejected = 'Rejected',
+}
+
+export interface ConnectWalletType {
+  buttonText?: string;
+  isLoading?: boolean;
+  isDisabled?: boolean;
+  icon?: IconType;
+  onClickConnectBtn?: MouseEventHandler<HTMLButtonElement>;
+}
+
+export interface ConnectedUserCardType {
+  walletIcon?: string;
+  username?: string;
+  icon?: ReactNode;
+}
+
+export interface FeatureProps {
+  title: string;
+  text: string;
+  href: string;
+}
+
+export interface ChainCardProps {
+  prettyName: string;
+  icon?: string;
+}
+
+export type CopyAddressType = {
+  address?: string;
+  walletIcon?: string;
+  isLoading?: boolean;
+  maxDisplayLength?: number;
+  isRound?: boolean;
+  size?: string;
+};
+
+export interface OptionBase {
+  variant?: string;
+  colorScheme?: string;
+  isFixed?: boolean;
+  isDisabled?: boolean;
+}
+
+export interface ChainOption extends OptionBase {
+  isDisabled?: boolean;
+  label: string;
+  value: string;
+  icon?: string;
+  chainName: string;
+  chainRoute?: string;
+}
+
+export type handleSelectChainDropdown = (value: ChainOption | null) => void;
+
+export interface ChangeChainDropdownType {
+  data: ChainOption[];
+  selectedItem?: ChainOption;
+  onChange: handleSelectChainDropdown;
+  chainDropdownLoading?: boolean;
+}
+
+export interface ChangeChainMenuType {
+  data: ChainOption[];
+  value?: ChainOption;
+  onClose?: () => void;
+  onChange: handleSelectChainDropdown;
+  innerRef?: RefObject<HTMLInputElement>;
+}
+
+export interface Any {
+  typeUrl: string;
+  value: Uint8Array;
+}
+
+export enum BondStatus {
```

Comment:
this should not be in our repo in `types.ts` — this should be imported from `interchain`